wrapper/src/main/java/software/amazon/jdbc/plugin/limitless/LimitlessConnectionPlugin.java

```diff
@@ -225,16 +283,16 @@ private Connection retryConnectWithLeastLoadedRouters(final List<HostSpec> limit
     throw new SQLException(Messages.get("LimitlessConnectionPlugin.noRoutersAvailableForRetry"), originalException);
```

Comment:
We also need to catch SQLException caused by strategy been unable to get a host and continue with our retry loop.

Reply:
Does it make sense to connect to initialHostSpec then?